common/src/types/store-types.ts

```diff
@@ -23,78 +23,59 @@ export type SubTaskWithoutIdOrder = Pick<SubTask, 'name' | 'complete' | 'inFocus
  */
 export type PartialSubTask = Partial<SubTaskWithoutIdOrder>;
 
-export type CommonTask<D> = {
-  readonly id: string;
-  readonly order: number;
-  readonly name: string; // Example: "Task 1 name"
-  readonly tag: string; // ID of the tag
-  readonly date: D;
-  readonly complete: boolean;
-  readonly inFocus: boolean; // Whether the task is in focus
-  readonly children: readonly SubTask[];
-};
-
-type FlexibleCommonTask = CommonTask<Date | RepeatMetaData>;
-
-export type OneTimeTask = CommonTask<Date> & {
+// TODO: refactor ical task to a separate category
```

Comment:
Are iCal tasks not also one time tasks?

Reply:
The hope is that iCal tasks are in a completely separate category, so that pattern matching mechanism will force us not to forget special-casing different behavior. e.g. disable edit.

For example, we might want something like this:
```typescript
function canBeEdited(metadata: TaskMetedata) {
    switch (metadata.type) {
        case 'ONE_TIME':
            return true;
        case 'I_CAL':
            // we will never forget to consider correctness for iCal tasks
            return false;
        // other cases ...
    }
}
```

Reply:
makes sense, the rest of the PR looks good to me